Add code formatting checking

Grillo-0 opened the pull request:
On the commit descriptions :^)

Grillo-0 committed:
Add .editorconfig file

EditorConfig helps maintain consistent coding styles for multiple
developers working on the same project across various editors and IDEs.
The EditorConfig project consists of a file format for defining coding
styles and a collection of text editor plugins that enable editors to
read the file format and adhere to defined styles. EditorConfig files
are easily readable and they work nicely with version control systems.

For more information go to: https://editorconfig.org/

Grillo-0 committed:
Add a .clang-format file

We need to standardize the code style of the project, the most popular
way to this for C/C++ projects is using clang-format. To do this, create
.clang-format file and reformat the code based on that.

Grillo-0 committed:
.github/build-vinumc: Add formatting checks

To facilitate the review process integrate clang-format into the CI.

JeanJPNM commented on .clang-format:
```diff
@@ -0,0 +1,12 @@
+BasedOnStyle: LLVM
+UseTab: Always
+IndentWidth: 8
+TabWidth: 8
+ContinuationIndentWidth: 8
```
Though we are only using tabs, 8 seems like an odd choice over the default 4

Grillo-0 replied:
I already have following the [kernel coding style](https://www.kernel.org/doc/html/latest/process/coding-style.html#indentation) on the project.

JeanJPNM commented:
It might be a good idea to mention that the project uses clang-format in the README.

Grillo-0 commented:
> It might be a good idea to mention that the project uses clang-format in the README.

Agreed, I also noticed that some code got a little weird after the format pass, thats why I switched the PR to draft, I will do some code changes to fix this.

Grillo-0 committed:
docs/CONTRIBUTING: Describe how to format the code with clang-format

Now that we use a tool for formatting our code describe how to use it
with meson.
